fbpic/particles/numba_methods.py

```diff
@@ -206,12 +201,18 @@ def deposit_field_numba(Fptcl, Fgrid,
     # Loop over all particles
     for ip in range(Ntot):
         # Loop over adjacent cells (given by shape order)
+        # Use helper variables `ir_corr` and `Sr_corr`, in order to avoid
+        # modifying ir and Sr in place. (This is not strictly necessary,
+        # but is just here as a safeguard.)
         for cell_index_r in range(ir.shape[0]):
             for cell_index_z in range(iz.shape[0]):
                 # Correct the guard cell index and sign
-                if ir[cell_index_r][ip] < 0:
-                    ir[cell_index_r][ip] = abs(ir[cell_index_r][ip]) - 1
-                    Sr[cell_index_r][ip] = sign_guards*Sr[cell_index_r][ip]
```

Comment:
I agree that modifying `ir` and `Sr` in-place works, but I would prefer not doing it, since it could be a somewhat unexpected side-effect of this function for users that don't read the function in details. 
Let me know if using `ir_corr` and `Sr_corr` here is fine with your.